packages/api/utils/requestValidation.ts

```diff
@@ -29,3 +33,98 @@ export function shapeValidationForPostRequest<T>(
 
   return new SetupSuccess<T>({ portfolioDraftId }, bodyResult as unknown as T);
 }
+export interface ClinValidationError {
+  clinNumber: string;
+  invalidParameterName: string;
+  invalidParameterValue: string;
+  validationMessage: ValidationMessage;
+}
+
+/**
+ * Check if the given Funding Step object passed business rule validation
+ *
+ * @param fs - the FundingStep object that has passed schema validation
+ * @returns an ValidationErrorResponse (by throwing an error to the middleware) if there are Clin validation errors
+ */
+
+export function businessRulesValidationForFundingStep(fs: FundingStep): ValidationErrorResponse | undefined {
```

Comment:
Can we use a verb for the function name? Perhaps `validateBusinessRulesForFundingStep`()  

Or, and this might be getting ahead of things, if we used a union type, more simply `validateBusinessRules(ps: PortfolioStep | fs: FundingStep | as: ApplicationStep): ValidationErrorResponse | undefined`

Reply:
Will change the name, and I like the second idea, probably will roll it into AT-6734.